core/src/main/java/org/elasticsearch/transport/TcpTransport.java

```diff
@@ -1577,4 +1583,14 @@ protected final void onChannelClosed(Channel channel) {
             }
         }
     }
+
+    /**
+     * Ensures this transport is still started / open
+     * @throws IllegalStateException if the transport is not started / open
+     */
+    protected final void ensureOpen() {
```

Comment:
I don't think it needs to hold a lock here. it's really best effort and can be used anywhere. Under lock it has different semantics but they are related to the context of the lock so I think we are good